uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -21,6 +22,27 @@ public function create($params, $validate = true) {
   }
 
   /**
+   * Deletes the LeaveRequest with the given $leaveRequestID, including all of
+   * its LeaveRequestDates and LeaveBalanceChanges
+   *
+   * @param int $leaveRequestID
+   */
+  public function delete($leaveRequestID) {
+    $leaveRequest = LeaveRequest::findById($leaveRequestID);
+
+    $transaction = new CRM_Core_Transaction();
+    try {
+      LeaveBalanceChange::deleteAllForLeaveRequest($leaveRequest);
+      LeaveRequestDate::deleteDatesForLeaveRequest($leaveRequest->id);
+      $leaveRequest->delete();
+
+      $transaction->commit();
+    } catch(Exception $e) {
+      $transaction->rollback();
+    }
```

Comment:
Unneeded. This is kinda like a wrapper around the BAO `del()` method, which doesn't return anything